### PR TITLE
[tf.data] move checkpoint tests to kernel tests part 5

### PR DESCRIPTION
**Pull request by kvignesh1420**
This PR is a continuation of #47668  and moves the checkpoint tests of:

1. batch
2. cache
3. concatenate
4. filter
5. fixed length record
6. flat map
7. interleave
8. padded batch
9. prefetch
10. range
11. rebatch
12. shard
13. shuffle
14. text line
15. tf record
16. unbatch and
17. zip
 dataset serialization tests to kernel tests

TEST LOG
```
//tensorflow/python/data/kernel_tests:batch_test                         PASSED in 15.5s
//tensorflow/python/data/kernel_tests:cache_test                         PASSED in 13.4s
//tensorflow/python/data/kernel_tests:concatenate_test                   PASSED in 3.7s
//tensorflow/python/data/kernel_tests:filter_test                        PASSED in 12.4s
//tensorflow/python/data/kernel_tests:fixed_length_record_dataset_test   PASSED in 3.6s
  Stats over 4 runs: max = 3.6s, min = 2.8s, avg = 3.3s, dev = 0.4s
//tensorflow/python/data/kernel_tests:flat_map_test                      PASSED in 15.7s
//tensorflow/python/data/kernel_tests:interleave_test                    PASSED in 24.4s
//tensorflow/python/data/kernel_tests:padded_batch_test                  PASSED in 8.0s
//tensorflow/python/data/kernel_tests:prefetch_test                      PASSED in 4.7s
//tensorflow/python/data/kernel_tests:range_test                         PASSED in 3.9s
//tensorflow/python/data/experimental/kernel_tests:rebatch_dataset_test  PASSED in 10.3s
//tensorflow/python/data/kernel_tests:shard_test                         PASSED in 13.4s
//tensorflow/python/data/kernel_tests:shuffle_test                       PASSED in 38.8s
//tensorflow/python/data/kernel_tests:text_line_dataset_test             PASSED in 5.6s
  Stats over 4 runs: max = 5.6s, min = 2.8s, avg = 4.2s, dev = 1.3s
//tensorflow/python/data/kernel_tests:tf_record_dataset_test             PASSED in 8.5s
  Stats over 4 runs: max = 8.5s, min = 2.7s, avg = 5.6s, dev = 2.9s
//tensorflow/python/data/kernel_tests:unbatch_test                       PASSED in 5.0s
//tensorflow/python/data/kernel_tests:zip_test                           PASSED in 5.0s
```

Also, this PR is a part of the larger cleanup as discussed in point 4 of https://github.com/tensorflow/tensorflow/pull/46761#issuecomment-770059963
cc: @jsimsa I went ahead with a bulk change to speed things up before the 2.5 branch cut.

### CODE REVIEW

File: tensorflow/python/data/kernel_tests/interleave_test.py [hunk 1]
```diff
@@ -355,6 +356,51 @@ def interleave_fn(x):
...
+            cycle_length, block_length, num_parallel_calls)
+
+  @combinations.generate(
+      combinations.times(
+          test_base.default_test_combinations(),
+          combinations.combine(
+              cycle_length=2,
+              block_length=[1, 3],
+              num_parallel_calls=[None, 1, 2])))
+  def testSerializationCore(self, cycle_length, block_length,
```
Comment (jsimsa) [inline tensorflow/python/data/kernel_tests/interleave_test.py:377]: rename to `testCore`

File: tensorflow/python/data/kernel_tests/range_test.py [hunk 2]
```diff
@@ -142,6 +150,90 @@ def testStopLessThanStartWithNegativeStep(self, output_type):
...
+        self.evaluate(variables.global_variables_initializer())
+        self.evaluate(init_op)
+        for i in range(start, break_point):
+          self.assertEqual(i, self.evaluate(get_next))
+        self.evaluate(save_op)
+        self.evaluate(restore_op)
+        for i in range(break_point, stop):
+          self.assertEqual(i, self.evaluate(get_next))
+        with self.assertRaises(errors.OutOfRangeError):
+          self.evaluate(get_next)
```
Comment (jsimsa) [inline tensorflow/python/data/kernel_tests/range_test.py:225]: Remove this as it is outdated.

**Review by jsimsa — APPROVED** [review]
Thank you, left a couple of minor comments.

File: tensorflow/python/data/kernel_tests/interleave_test.py (hunk 1 above)
Comment (kvignesh1420) [reply tensorflow/python/data/kernel_tests/interleave_test.py]: addressed in the commit

File: tensorflow/python/data/kernel_tests/range_test.py (hunk 2 above)
Comment (kvignesh1420) [reply tensorflow/python/data/kernel_tests/range_test.py]: removed the methods, imports and extra build deps. Test log after change.
```
//tensorflow/python/data/kernel_tests:range_test                         PASSED in 3.2s
```

**Comment by kvignesh1420** [comment]
@jsimsa made the changes!

**Review by jsimsa — APPROVED** [review]

**Review by jsimsa — APPROVED** [review]

**Comment by kvignesh1420** [comment]
@jsimsa I was working on the next PR and was able to move all the tests except these two:
- [checkpoint_input_pipeline_hook_test.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/experimental/kernel_tests/serialization/checkpoint_input_pipeline_hook_test.py)
- [serialization_integration_test.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/experimental/kernel_tests/serialization/serialization_integration_test.py)

Where do you suggest that we put them?

**Comment by jsimsa** [comment]
> @jsimsa I was working on the next PR and was able to move all the tests except these two:
> 
> * [checkpoint_input_pipeline_hook_test.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/experimental/kernel_tests/serialization/checkpoint_input_pipeline_hook_test.py)
> * [serialization_integration_test.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/experimental/kernel_tests/serialization/serialization_integration_test.py)
> 
> Where do you suggest that we put them?

Both can go to `experimental/kernel_tests` (i.e. move one level up in the directory hierarchy). The second test should also be renamed to `make_saveable_from_iterator_test.py`.

**Comment by kvignesh1420** [comment]
@jsimsa thanks. I made the change and will raise the PR once this is merged.

**Review by gbaned — APPROVED** [review]

**Pull request merged**